lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -123,6 +136,23 @@ def update_source(status)
           logger.availability_check("Failed to update Source id:#{source_id} - #{e.message}", :error)
         end
 
+        def update_source_by_kafka(status)
+          client = ManageIQ::Messaging::Client.open(:protocol => :Kafka, :host => messaging_host, :port => messaging_port)
```

Comment:
Also, for this client I think we should memoize it into a variable, that way we don't instantiate the messaging client over and over again.

Reply:
@pkomanek you can use `cache_with_timeout` method (https://github.com/RedHatInsights/topological_inventory-sync/blob/ea0730de9a13deefad79af38b01dd4c3a007944a/lib/topological_inventory/sync/sources_sync_worker.rb#L163)